### ISSUE TITLE
Add a Custom Titlebar toolbar for actions

### ISSUE THREAD

**Issue by felipetesc** [zed-industries/zed#15024]
### Check for existing issues

- [X] Completed

### Describe the feature

Create a custom toolbar with buttons placed on the middle of the titlebar, between the menu's dropdown button and the dropdown with the items "settings", "key bindings", "themes" and "extensions" .  Each button calls a command inside a json file located at the toot of the project folder. If a given command can be enabled or disabled the button becames highlighted.

Visual example:
![sample_actions_bar](https://github.com/user-attachments/assets/8ffbeb1b-88c5-4284-ba89-2bceadaebf28)

Zed with custom titlebar, just an example: 

![example_actions_panel_toolbar](https://github.com/user-attachments/assets/f023a766-4c35-4a6c-9890-9565614c14ae)

Since every button can have a related action, users can customize the icon. Rust can use std::process::Command for each action inside the json.

A config file can have this fields and can  have this form, i.e. :
```json
{
    "commands":[
        {
            "name": "Run Release",
            "cmd": "cargo",
            "args": ["run", "--release"],
            "description": "Build and run",
            "shortcut": "F5",
            "icon": "${ProjectFolder}/.zed/icons/run.svg"
        },
        {
            "name": "Run Debug",
            "cmd": "cargo",
            "args": ["run"],
            "description": "Build and run",
            "shortcut": "F6",
            "icon": "${ProjectFolder}/.zed/icons/run.svg"
        },
    ]
        
    
}
```

### If applicable, add mockups / screenshots to help present your vision of the feature

Example of custom actions panel (VSCode uses as a debug panel):
![toolbar](https://github.com/user-attachments/assets/8a30a098-7370-4c9f-ab45-f9683246bdda)
Source: https://code.visualstudio.com/docs/editor/debugging

Another example:
<img width="420" alt="187667425-37058429-50c6-47d7-898d-c0af0d163b06" src="https://github.com/user-attachments/assets/5708ee8c-b1ab-4be9-8009-f11907913fb5">

Fleet has this custom debug action toolbar:
![rerun_button_in_a_floating_terminal](https://github.com/user-attachments/assets/a952dbec-161d-43f2-aec8-3e339939da18)

XCode has this:

![xcode_toolsbar](https://github.com/user-attachments/assets/642784e3-cc9a-461f-add6-f2a4653f13cb)

**Comment by JosephTLyons** [comment]
Hey @felipetesc - I realize this issue is asking for more than just this, but Zed does already support tasks:

- https://zed.dev/docs/tasks

In case you haven't found out about these yet.

**Comment by felipetesc** [comment]
> Hey @felipetesc - I realize this issue is asking for more than just this, but Zed does already support tasks:
> 
>     * https://zed.dev/docs/tasks
> 
> 
> In case you haven't found out about these yet.

Yes, Zed's tasks are related to my feature request, and they have almost the same fields I exemplified on the top of the current issue with the given json file. So, I'm guessing a little bit of rework to connect a task to custom panel with custom icon could do the trick.
